### PR TITLE
feat(argo-cd): Add templating of ingress extra paths and rules

### PR DESCRIPTION
**Pull request by pdrastil**
Resolves: https://github.com/argoproj/argo-helm/issues/2527

Added `tpl` where user might need to reference backend service name.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -2125,15 +2129,17 @@ server:
...
-      #   http:
-      #       path: /
+      # - http:
+      #     paths:
+      #     - path: /
+      #       pathType: Prefix
       #       backend:
       #         service:
-      #           name: example-svc
+      #           name: '{{ include "argo-cd.server.fullname" . }}'
```
[the remark this answers is not in the document]
Comment (pdrastil) [reply charts/argo-cd/values.yaml:2140]: It works with Helm 3.14.1. You can try it.

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply charts/argo-cd/values.yaml:2148]: Alright it works 👍 tested it with helm v3.13.2.

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -2028,15 +2029,17 @@ server:
 
     # -- Additional ingress rules
     # @default -- `[]` (See [values.yaml])
+    ## Note: Supports use of custom Helm templates
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:2032]: Can we move this before the @default line so it is present inside the README?
Maybe we can also use the syntax of Bitnami charts
`(evaluated as a template)`?

**Review by mkilchhofer — COMMENTED** [review]
LGTM but maybe we should add it to the README inside the parameter table.

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (pdrastil) [reply charts/argo-cd/values.yaml:2049]: @mkilchhofer We have this note in other places such as `extraContainers`. Do you want to change that as well?

**Review by yu-croco — APPROVED** [review]

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (mkilchhofer) [reply charts/argo-cd/values.yaml:2053]: Ah, I forgot that 😄 
Lets continue then 👍

**Pull request merged**